lib/array.rb

```diff
@@ -0,0 +1,11 @@
+class Array
+
+	def my_inject(argument = 0)
+		return nil if self == []
```

Comment:
1) you could use `empty?` rather than self == []

2) inject doesn't return nil if you pass a value to inject, it will return the value you pass to the method